ghcide/src/Development/IDE/Monitoring/OpenTelemetry.hs

```diff
@@ -0,0 +1,31 @@
+module Development.IDE.Monitoring.OpenTelemetry (monitoring) where
+
+import           Control.Concurrent.Async         (Async, async, cancel)
+import           Control.Monad                    (forever)
+import           Data.IORef.Extra                 (atomicModifyIORef'_,
+                                                   newIORef, readIORef)
+import           Data.Text.Encoding               (encodeUtf8)
+import           Debug.Trace.Flags                (userTracingEnabled)
+import           Development.IDE.Types.Monitoring (Monitoring (..))
+import           OpenTelemetry.Eventlog           (mkValueObserver, observe)
+import           System.Time.Extra                (Seconds, sleep)
+
+-- | Dump monitoring to the eventlog using the Opentelemetry package
+monitoring :: IO Monitoring
+monitoring
+  | userTracingEnabled = do
+    actions <- newIORef []
+    let registerCounter name read = do
+            observer <- mkValueObserver (encodeUtf8 name)
+            let update = observe observer . fromIntegral =<< read
+            atomicModifyIORef'_ actions (update :)
+        registerGauge = registerCounter
```

Comment:
I guess it's fine that we use the same implementation? I'm not sure whether anything cares about this distinction.

Reply:
This is just packaging the previously existing open telemetry monitoring, which didn't make this distinction,  to fit in the new abstraction